astropy/coordinates/tests/test_earth.py

```diff
@@ -410,3 +437,236 @@ def test_info():
+        assert quantity_allclose(s1.lat, [5, 6] * u.degree)
+        assert quantity_allclose(s1.height, [10, 10] * u.m)
+
+    def test_broadcasting_mismatch(self):
+
+        with pytest.raises(ValueError) as exc:
+            s1 = WGS84GeodeticRepresentation(lon=[8, 9, 10] * u.hourangle,
+                                         lat=[5, 6] * u.deg,
+                                         height=[1, 2] * u.km)
+        assert exc.value.args[0] == "Input parameters lon, lat, and height cannot be broadcast"
+
+    def test_readonly(self):
+
+        s1 = WGS84GeodeticRepresentation(lon=8 * u.hourangle,
+                                     lat=5 * u.deg,
+                                     height=1. * u.km)
+
+        with pytest.raises(AttributeError):
+            s1.lon = 1. * u.deg
+
+        with pytest.raises(AttributeError):
+            s1.lat = 1. * u.deg
+
+        with pytest.raises(AttributeError):
+            s1.height = 1. * u.km
+
+    def test_getitem_len_iterable(self):
+
+        s = WGS84GeodeticRepresentation(lon=np.arange(10) * u.deg,
+                                    lat=-np.arange(10) * u.deg,
+                                    height=1 * u.m)
+
+        s_slc = s[2:8:2]
+
+        assert quantity_allclose(s_slc.lon, [2, 4, 6] * u.deg)
+        assert quantity_allclose(s_slc.lat, [-2, -4, -6] * u.deg)
+        assert quantity_allclose(s_slc.height, [1, 1, 1] * u.m)
+
+        assert len(s) == 10
+        assert isiterable(s)
+
+    def test_getitem_len_iterable_scalar(self):
+
+        s = WGS84GeodeticRepresentation(lon=1 * u.deg,
+                                    lat=-2 * u.deg,
+                                    height=300 * u.m)
+
+        with pytest.raises(TypeError):
+            s_slc = s[0]
+        with pytest.raises(TypeError):
+            len(s)
+        assert not isiterable(s)
+
+    def test_setitem(self):
+        s = WGS84GeodeticRepresentation(lon=np.arange(5) * u.deg,
+                                    lat=-np.arange(5) * u.deg,
+                                    height=1 * u.m)
+        s[:2] = WGS84GeodeticRepresentation(lon=10.*u.deg, lat=2.*u.deg,
+                                        height=5.*u.m)
+        assert quantity_allclose(s.lon, [10, 10, 2, 3, 4] * u.deg)
+        assert quantity_allclose(s.lat, [2, 2, -2, -3, -4] * u.deg)
+        assert quantity_allclose(s.height, [5, 5, 1, 1, 1] * u.m)
+
+    def test_nan_distance(self):
+        """ This is a regression test: calling represent_as() and passing in the
+            same class as the object shouldn't round-trip through cartesian.
+        """
+
+        sph = WGS84GeodeticRepresentation(1*u.deg, 2*u.deg, np.nan*u.m)
+        new_sph = sph.represent_as(WGS84GeodeticRepresentation)
+        assert quantity_allclose(new_sph.lon, sph.lon)
+        assert quantity_allclose(new_sph.lat, sph.lat)
+
+        # dif = SphericalCosLatDifferential(1*u.mas/u.yr, 2*u.mas/u.yr,
```

Comment:
Probably good to put a TODO on top that we do not yet have differentials (and then remove the commented out code here)

Reply:
Done!